tests/handlers/test_presence.py

```diff
@@ -505,7 +507,7 @@ def test_last_active(self):
         self.assertEqual(state, new_state)
 
 
-class PresenceHandlerTestCase(unittest.HomeserverTestCase):
+class PresenceHandlerTestCase(BaseMultiWorkerStreamTestCase):
```

Comment:
Nitpicking, but I'm wondering about the unnecessary load this might bring (if any) to other tests in this test case that don't care about workers. Though that's probably too small of an impact to matter much.

Reply:
Removing the newly added `test_set_presence_from_syncing_keeps_busy` and benchmarking the existing test class with `hyperfine`, we can see that inheriting from `BaseMultiWorkerStreamTestCase` does incur a small performance penalty.

Inheriting from `BaseMultiWorkerStreamTestCase`:

```
$ SYNAPSE_TEST_LOG_LEVEL=INFO SYNAPSE_POSTGRES=1 SYNAPSE_POSTGRES_USER=synapse_us
er SYNAPSE_POSTGRES_HOST=127.0.0.1 SYNAPSE_POSTGRES_PASSWORD=synapse hyperfine "python -m twisted.trial tests.handlers.test_presence.
PresenceHandlerTestCase"
Benchmark 1: python -m twisted.trial tests.handlers.test_presence.PresenceHandlerTestCase
  Time (mean ± σ):      9.222 s ±  0.253 s    [User: 2.886 s, System: 0.099 s]
  Range (min … max):    8.724 s …  9.593 s    10 runs
```

Inheriting from `unittest.HomeserverTestCase`:

```
$ SYNAPSE_TEST_LOG_LEVEL=INFO SYNAPSE_POSTGRES=1 SYNAPSE_POSTGRES_USER=synapse_user SYNAPSE_POSTGRES_HOST=127.0.0.1 SYNAPSE_POSTGRES_PASSWORD=synapse hyperfine "python -m twisted.trial tests.handlers.test_presence.PresenceHandlerTestCase"
Benchmark 1: python -m twisted.trial tests.handlers.test_presence.PresenceHandlerTestCase
  Time (mean ± σ):      8.826 s ±  0.243 s    [User: 2.445 s, System: 0.094 s]
  Range (min … max):    8.218 s …  9.129 s    10 runs
```

For roughly `8.826s` - `9.222s` = `396ms` total overhead, or `44ms` per each tests (9 total). Splitting the test case is fairly trivial, though we would need to duplicate/convert `_set_presencestate_with_status_msg` to a static method (or move it into test utilities?). Probably not worth it IMO, but worth checking!